qc_tests/streaks.py

```diff
@@ -4,28 +4,33 @@
 
    Checks for replication of 
      1. checks for consecutive repeating values
-     2. checks if one year has more repeating strings than expected [Not yet implemented]
+     2. checks if one year has more repeating streaks than expected
      3. checks for repeats at a given hour across a number of days [Not yet implemented]
-     4. checks for repeats for whole days - all 24 hourly values [Not yet implemented]
+     4. checks for repeats for whole days - all 24 hourly values
 
    Thresholds now determined dynamically
 """
 #************************************************************************
 import copy
 import numpy as np
+import datetime as dt
 import logging
 logger = logging.getLogger(__name__)
+from typing import Optional
 
 import qc_utils as utils
 
+MIN_STREAK_LENGTH_FOR_EXCESS_FREQUENCY = 10
```

Comment:
Are these consecutive hours or consecutive obs? So obs could be 8 hourly therefor on 2 obs may be needed to identify a 10 hour streak compared to 10 obs for hourly data..